### PR TITLE
fix: wrapper function for hapi route & plugins

### PR DESCRIPTION
**Pull request by jk1z**
## Which problem is this PR solving?

- Plugin wrapping issue https://github.com/open-telemetry/opentelemetry-js-contrib/issues/218
- Route handler wrapping issue https://github.com/open-telemetry/opentelemetry-js-contrib/issues/220

## Short description of the changes

- Fix wrapping issues in plugin and route handler

### CODE REVIEW

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:371]: simplify this to
`route.options?.handler`

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:159]: simplify this to
```
pluginInput.plugin?.plugin ?? pluginInput.plugin
```

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:152]: simplify this to
```
pluginObj.plugin?.plugin ?? pluginInput.plugin
```

File: package.json
Comment (obecny) [inline package.json:71]: was this changed accidentally?

File: plugins/node/opentelemetry-hapi-instrumentation/test/hapi-plugin.test.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/test/hapi-plugin.test.ts:333]: can you please refactor this and next test into more readable format

```
describe('when handler is in options', ()=> {
  it('should ...
})
describe('when handler is in export.plugin', ()=> {
  it('should ...
})
```
etc.

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (jk1z) [reply plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:395]: @obecny Just wondering if that will affect the backwards compatibility on previous node version? Since the `?` is a new feature in ES2020 which is only fully supported in node 14

Comment (obecny) [reply plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:395]: you are writing in typescript which is then compiled

Comment (jk1z) [reply plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:395]: ahhh ok right.

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (jk1z) [reply plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:159]: I think I can simplify it with acquiring handler part as well

File: package.json
Comment (jk1z) [reply package.json:71]: I will revert it back. My IDE is doing some weird things

**Comment by jk1z** [comment]
@obecny I have pushed up the changes. Can I get another review please?

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:383]: Nit: to keep the same order as in line 364 I think it should be
```
if (route.options?.handler) {
  route.options.handler = newHandler
} else {
  route.handler = newHandler;
}
```

Comment (jk1z) [reply plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts]: Yea sure, I will make the change

File: plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts
Comment (obecny) [inline plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts:385]: Nit: only `else` would be enough due to line 364 and 365

**Review by obecny — APPROVED** [review]
lgtm

**Comment by jk1z** [comment]
Hi @obecny Just wondering what's the ETA on this getting merged? Our org is using the forked version of the hapi plugin

**Comment by obecny** [comment]
@open-telemetry/javascript-approvers ^^

**Review by mwear — APPROVED** [review]

**Comment by dyladan** [comment]
@jk1z I can't merge this until you update the branch (or check the box that allows maintainers to update your branch 😄 )

**Review by dyladan — APPROVED** [review]

**Comment by jk1z** [comment]
> @jk1z I can't merge this until you update the branch (or check the box that allows maintainers to update your branch 😄 )

That's done. 🙏

**Pull request merged**
